pkg/controller/operator/master/resources/kubermatic/common.go

```diff
@@ -97,7 +97,12 @@ func IngressReconciler(cfg *kubermaticv1.KubermaticConfiguration) reconciling.Na
 			if i.Annotations == nil {
 				i.Annotations = make(map[string]string)
 			}
-			i.Annotations["kubernetes.io/ingress.class"] = cfg.Spec.Ingress.ClassName
+			i.Annotations["kubernetes.io/ingress.class"] = cfg.Spec.Ingress.ClassNam
+
+			// NGINX ingress annotations to avoid timeout of websocket connections after 1 minute.
+			// Needed for Web Terminal feature, for example.
+			i.Annotations["nginx.ingress.kubernetes.io/proxy-read-timeout"] = "3600" // 1 hour
+			i.Annotations["nginx.ingress.kubernetes.io/proxy-send-timeout"] = "3600" // 1 hour
```

Comment:
> NGINX ingress annotations to avoid timeout of websocket connections after 1 minute

To avoid timeout after 1 minute, we are setting the timeout to 1 hour?

Reply:
1 hour if no "read" or "send" activity happens.

Reply:
1 hour seems way too much to me but maybe it's fine. @rastislavs WDYT?

Reply:
1 hour seems way too much to me but maybe it's fine. @rastislavs WDYT?

Reply:
I think 3600 is fine. It also matches with similar timeout [that we have on the nodeport-proxy](https://github.com/kubermatic/kubermatic/blob/main/pkg/resources/nodeportproxy/nodeportproxy.go#L441).

Reply:
Fair point, lgtm then :D